lib/repository/villager_details.dart

```diff
@@ -0,0 +1,14 @@
+import 'package:acnhpedia/api_key.dart';
+import 'package:acnhpedia/networking/api_base_helper.dart';
+import 'package:acnhpedia/models/villager_model.dart';
+
+class VillagerDetailRepository {
+  final String _apiKey = api_key;
+
+  ApiBaseHelper _helper = ApiBaseHelper();
+
+  Future<villager> fetchVillagerData(String villager_name) async {
+    final response = await _helper.get("villagers?x-api-key=$_apiKey");
```

Comment:
api key should be handled within the get function itself. otherwise you will have to mention API key everywhere you call a get/post method. make it generic and abstract